### PR TITLE
Added StealAppLease API and HTTP API

### PR DESCRIPTION
**Pull request by amdeel**
Added StealAppLease API and HTTP API to support app lease swap without turning off a function instance.

Extension side work for this issue: https://github.com/Azure/azure-functions-durable-extension/issues/1503

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs [hunk 1]
```diff
@@ -298,5 +298,11 @@ public interface IDurableOrchestrationClient
         /// <returns>A task that completes when the orchestration is started. The task contains the instance id of the started
         /// orchestratation instance.</returns>
         Task<string> RestartAsync(string instanceId, bool restartWithNewInstanceId = true);
+
+        /// <summary>
+        ///  Steals AppLease if this app doesn't already have it. To use this, must be using the AppLease feature by setting UseAppLease to true in host.json.
+        /// </summary>
+        /// <returns>A task that completes when the steal app message is written to storage.</returns>
```
Comment (davidmrdavid) [inline src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs:305]: Just for clarity: is writing the steal message to storage the same as completing the process of stealing the lease, or is it just a starting step in that process?

**Review by davidmrdavid — COMMENTED** [review]
Just a tiny comment, thanks!

File: src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs (hunk 1 above)
Comment (ConnorMcMahon) [reply src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs:305]: We probably want to be explicit in the documentation saying that this kicks off the process to cause this application to steal the app lease.

**Review by davidmrdavid — APPROVED** [review]
LGTM so long as the CI goes green!

**Pull request merged**